### PR TITLE
Enable API review gen in CI for Go

### PR DESCRIPTION
**Pull request by praveenkuttappan**
Add step in CI pipeline to save package properties

### CODE REVIEW

File: eng/pipelines/templates/steps/build-test.yml [hunk 1]
```diff
@@ -82,6 +82,36 @@ steps:
...
+
+  - template: /eng/common/pipelines/templates/steps/set-default-branch.yml
+  
+  - task: Powershell@2
+    inputs:
+      filePath: $(Build.SourcesDirectory)/eng/scripts/Create-ApiReview.ps1
+      arguments: >
+        -ServiceDirectory ${{parameters.ServiceDirectory}}
+        -OutPath $(Build.ArtifactStagingDirectory)
+        -ApiviewUri "$(azuresdk-apiview-uri)"
```
Comment (benbp) [inline eng/pipelines/templates/steps/build-test.yml:104]: @praveenkuttappan do you know if all pipelines already have access to this or if I need to do a pipeline gen update to propagate this variable?

File: eng/scripts/Language-Settings.ps1 [hunk 2]
```diff
@@ -130,3 +130,17 @@ function SetPackageVersion ($PackageName, $Version, $ReleaseDate, $PackageProper
     -ReleaseDate $ReleaseDate `
     -ReplaceLatestEntryTitle $ReplaceLatestEntryTitle
 }
+
+
+function Find-Go-Artifacts-For-Apireview($ArtifactPath, $PackageName)
+{
+  $artifact = Get-ChildItem -Path (Join-Path $ArtifactPath $PackageName) -Filter "*.gosource"
```
Comment (benbp) [inline eng/scripts/Language-Settings.ps1:137]: Does this need `-Recurse` for nested paths or no?

**Review by benbp — APPROVED** [review]

**Comment by benbp** [comment]
Partially implements #17594 (just not blocking to start).

File: eng/scripts/Language-Settings.ps1 (hunk 2 above)
Comment (praveenkuttappan) [reply eng/scripts/Language-Settings.ps1:137]: Not really. I will be surprised if it happens to be in another sub directory and we can take a look.

**Comment by benbp** [comment]
FYI @chlowell @RickWinter

File: eng/pipelines/templates/steps/build-test.yml (hunk 1 above)
Comment (praveenkuttappan) [reply eng/pipelines/templates/steps/build-test.yml:104]: It's already part of our core variable group that gets linked to each pipeline created by pipeline generator.

File: eng/scripts/Create-ApiReview.ps1 [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    [Parameter(Mandatory=$True)]
+    [string] $SourceBranch,
+    [Parameter(Mandatory=$True)]
+    [string] $DefaultBranch,
+    [Parameter(Mandatory=$True)]
+    [string] $ConfigFileDir
+)
+
+
+Write-Host "$PSScriptRoot"
```
Comment (weshaggard) [inline eng/scripts/Create-ApiReview.ps1:21]: I assume this was just for debugging.

**Pull request merged**